src/Microsoft.Framework.Configuration.Abstractions/IConfigurationBuilder.cs

```diff
@@ -13,6 +13,6 @@ public interface IConfigurationBuilder
 
         IConfigurationBuilder Add(IConfigurationSource configurationSource);
 
-        IConfiguration Build();
+        IConfigurationRoot Build();
```

Comment:
We need to remember changing the type of `Startup.Configuration` to this in the default template. I suspect no tests will break if we don't but customers will loose access to `Reload()`. It will be good to get more eyes that care about the template scrutinize the new name.